hbw/app/services/hbw/with_definitions.rb

```diff
@@ -19,12 +19,23 @@ def process_definitions
     def zip(tasks, definitions, variables)
       tasks.map do |task|
         new(task.merge(
-              'processDefinition' => definitions.find { |d| d.id == task.fetch('processDefinitionId') },
+              'processDefinition' => fetch_definition(definitions, task.fetch('processDefinitionId')),
               'variables' => HBW::Variable.wrap(variables.select { |var| var.fetch('processInstanceId') == task.fetch('processInstanceId') })
             ))
       end
     end
 
+    def fetch_definition(definitions, id)
+      definition = definitions.find { |d| d.id == id }
+
+      if definition.nil?
+        Rails.cache.delete(:process_definitions)
+        fetch_definition(process_definitions, id)
```

Comment:
I'm not sure about recursion here. Think it's meaningless to check this thing twice o more.
I think if we have task without process definition it's not much chances it would appear.